modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -69,6 +69,10 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
             {{#isPrimitiveType}}
             requestContext.setQueryParam("{{baseName}}", ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}"));
             {{/isPrimitiveType}}
+            {{#isEnumRef}}
+            requestContext.setQueryParam("{{baseName}}", ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}"));
+            {{/isEnumRef}}
+            {{^isEnumRef}}
             {{^isPrimitiveType}}
```

Comment:
Suggest swapping the conditions:
```suggestion
            {{^isPrimitiveType}}
            {{#isEnumRef}}
            requestContext.setQueryParam("{{baseName}}", ObjectSerializer.serialize({{paramName}}, "{{{dataType}}}", "{{dataFormat}}"));
            {{/isEnumRef}}
            {{^isEnumRef}}
```

This way, nothing will break if in future we change the codegen to mark referenced enums as "primitive".

Currently, if a ref'd enum is also marked as primitive, we render both L70 and L73. With the suggested change, we guarantee to only render either L70 or L73, but never the both

Reply:
@amakhrov , thanks for the suggestion!
I had to change the order to make it work (error below), which highlight the need to add a test or a sample. 
`Caused by: com.samskivert.mustache.MustacheParseException: Section close tag with mismatched open tag 'isPrimitiveType' != 'isEnumRef' @ line 93`

Reply:
@IvanVas the error seems because an opened section was not matched correctly. can you try again, making sure that a `{{#bla}}` or `{{^bla}}` is matched with a `{{/bla}}`

Reply:
> @IvanVas the error seems because an opened section was not matched correctly. can you try again, making sure that a `{{#bla}}` or `{{^bla}}` is matched with a `{{/bla}}`

Thanks! 
I fixed it, the current code state has the working version.